### ISSUE TITLE
[Feature Request] Better handling save file updates

### ISSUE THREAD

**Issue by AkazaRenn** [Ryujinx/Ryujinx#5472]
### Overview

Currently, whenever there's an update on the save file, Ryujinx wipes out the whole folder of that game and re-add all files. I would like it to be updated to only modify the folder/file that is updated (e.g. specific slot of save, newly added photo in album). 

### Smaller details

Take TOTK for example, whenever there's a save update, all folders and files in the `0` folder gets updates, you can tell that by the entry `Date modified` in the following screenshot:
![image](https://github.com/Ryujinx/Ryujinx/assets/8340896/e8299fca-d436-450f-904a-87dd5d5e078f)

In Yuzu's implementation, in my case of using `slot_05`, only files in `slot_05` folder will be modified, and `slot_05` folder itself remains untouched, it can also be told of the `Date modified` entry.

It will also benefit a lot on `album` and `picturebook`, given there are quite some files inside, and OneDrive has to keep deleting and uploading all the pictures, even when there's only an addition.

### Nature of request

This usually is fine, but I've made a symlink of the save folder to OneDrive to have it backed up, and it caused some trouble for OneDrive to correctly handle file updates. I understand that it might be the issue of OneDrive, but it could still be good to have Ryujinx handling it better. I have compared the behaviour between Ryujinx and Yuzu, and found Yuzu only updates the one save file, looking much better.

Sometimes OneDrive is even causing the game to crash. I'm not sure about the root cause, but in OneDrive it shows `CacheStorageKey.dat` is occupied. Every time it happens, I have to stop sync and restart sync to even get Ryujinx start again.

### Why would this feature be useful?

First OneDrive should work with Ryujinx much better. With correct modification time on the folders, we can also pick the correct save file much more easily. It can also result in less write to the disk, extending the disk life, not by much though.

**Comment by AcK77** [comment]
The way we handle savedata files is nearly the same as the Switch OS. There is nothing we can do here. Maybe @Thealexbarney want to say something about that ?

**Comment by Thealexbarney** [comment]
> The way we handle savedata files is nearly the same as the Switch OS. There is nothing we can do here. Maybe @Thealexbarney want to say something about that ?

This is correct. The way we handle directory savedata is the same way the Switch OS handles it. Copying the entire save directory is needed because modifying savedata is transactional, meaning none of the file changes are actually applied until the game "commits" those changes.

Among other things, this can help protect against things like crashes happening while a game is writing to the savedata. It would also handle situations where a game writes to the savedata but then backs out of the changes by closing the savedata without committing it.

If a dev really wanted to, they could delete all the files in the savedata and then close it without committing. Ryujinx and an actual Switch would handle this properly, but Yuzu would end up with the contents of the savedata being completely deleted.
